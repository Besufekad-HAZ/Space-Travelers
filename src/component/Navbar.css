.header {
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: #ffffff;
  margin-bottom: 2%;
}

#icon_logo {
  width: 55px;
  height: auto;
  margin-right: 10px;
}

.header .logo:hover {
  color: #0153c5;
}

.header nav {
  display: flex;
  align-items: center;
  padding: 1rem 2.5rem;
}

#nav_header {
  margin-left: 1%;
  margin-right: 22%;
}

nav ul {
  display: flex;
  gap: 1rem;
  list-style: none;
  font-weight: 500;
  text-decoration: none;
}

nav ul li a {
  color: rgb(0, 138, 251);
  font-size: 1.1rem;
  text-decoration: none;
}

ul a.active {
  text-decoration: underline;
  color: #2266c6;
}
