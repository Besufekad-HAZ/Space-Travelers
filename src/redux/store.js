// import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import {
  legacy_createStore as createStore,
  combineReducers,
  applyMiddleware,
} from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';
import missionsSlice from './missions/slice-Missions';

const reducers = combineReducers({
  rockets: rocketsSlice,
  missions: missionsSlice,
});

const store = createStore(reducers, applyMiddleware(thunk));
export default store;
