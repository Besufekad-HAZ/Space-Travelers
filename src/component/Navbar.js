import React from 'react';
import './Navbar.css';
import { NavLink } from 'react-router-dom';

const Navbar = () => (
  <div>
    <header className="header">
      <nav className="nav-bar">
        <div className="logo">
          Space Travelers Hub
        </div>
        <ul>
          <li><NavLink to="/">Rockets</NavLink></li>
          <li><NavLink to="/MyProfile">My Profile</NavLink></li>
        </ul>
      </nav>
    </header>
  </div>
);

export default Navbar;
