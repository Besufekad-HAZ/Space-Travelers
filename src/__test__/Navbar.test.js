import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Navbar from '../component/Navbar';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('The Navbar component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Navbar />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
