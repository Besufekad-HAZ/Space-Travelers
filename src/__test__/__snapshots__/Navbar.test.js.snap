// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar component renders correctly 1`] = `
<div>
  <header
    className="header"
  >
    <nav
      className="nav-bar"
    >
      <img
        alt="planet"
        id="icon_logo"
        src="planet.png"
      />
      <h2
        id="nav_header"
      >
        Space travelers' Hub
      </h2>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            className=""
            href="/Missions"
            onClick={[Function]}
          >
             Missions 
          </a>
        </li>
        <li>
          <a
            className=""
            href="/MyProfile"
            onClick={[Function]}
          >
             | My Profile
          </a>
        </li>
      </ul>
    </nav>
    <hr
      className="line"
    />
  </header>
</div>
`;
